app/controllers/admins_controller.rb

```diff
@@ -0,0 +1,5 @@
+class AdminsController < ApplicationController
+ before_action :authenticate_admin!,only: [:welcome]
+	def welcome
```

Comment:
If there is only one method require before filter then you can set it like
`before_action :authenticate_admin!,only: :welcome`
 no need to mention in array